### PR TITLE
auth filter added

### PR DESCRIPTION
**Pull request by urviljoshi**

### CODE REVIEW

File: src/filters/authfilter.bal [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    public function __init() {
+        self.clientEndpoint = new (config:getAsString("mosip.auth.base_url"));
+    }
+    *http:RequestFilter;
+
+    public function filterRequest(http:Caller caller, http:Request request,
+        http:FilterContext context) returns boolean {
+        if (config:getAsString("mosip.auth.filter_status", "disable") == "enable") {
+            http:Response errorsResponse = new;
+            if (request.getQueryParamValue("hub.mode") == "publish") {
```
Comment (vishwa-vyom) [inline src/filters/authfilter.bal:21]: Are we checking only for publish, for subscription are we failing the auth?

File: src/filters/authfilter.bal [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                    (http:Cookie cookie) returns boolean {
+                    return cookie.name == "Authorization";
+                });
+                if (authCookie.length() > 0) {
+                    string? authToken = authCookie[0].value;
+                    if (authToken is string) {
+                        http:Request req = new;
+
+                        req.addHeader("Cookie", "Authorization=".concat(authToken));
+                        log:printInfo("authtoken" + authToken);
```
Comment (vishwa-vyom) [inline src/filters/authfilter.bal:33]: Can we make this debug log, since it is a sensitive info?

File: src/filters/models.bal [hunk 3]
```diff
@@ -0,0 +1,23 @@
+public type ServiceError record {
```
Comment (vishwa-vyom) [inline src/filters/models.bal:1]: Are we using the below models anywhere?

File: src/filters/authfilter.bal (hunk 2 above)
Comment (urviljoshi) [reply src/filters/authfilter.bal:33]: sure

File: src/filters/models.bal (hunk 3 above)
Comment (urviljoshi) [reply src/filters/models.bal:1]: will remove this.

File: src/filters/authfilter.bal (hunk 1 above)
Comment (urviljoshi) [reply src/filters/authfilter.bal:21]: Let me add a condition for Sub also. Discussion of sub was unclear when we discussed with that team but let me add one condition and will add the same logic there.

Comment (vishwa-vyom) [reply src/filters/authfilter.bal]: @urviljoshi We should separate permission/role for publishing and subscribing to a topic. Someone authorized to only subscribe should be allowed to publish

**Pull request closed without merging**